### PR TITLE
Rework locked and add dmLocked

### PR DESCRIPTION
**Pull request by cyelis1224**
Changed locked to only affect player control and added dmLocked to also lock tokens for DM.

see video:

https://user-images.githubusercontent.com/82621530/145080530-b2336c59-852f-46c6-b3d3-6955e4eddbd8.mp4

### CODE REVIEW

**Comment by cyelis1224** [comment]
fixed name scheme and locking issues.

File: Token.js [hunk 1]
```diff
@@ -108,6 +108,7 @@ class Token {
 	}
 	rotate(newRotation) {
 		if (this.options.locked) return; // don't allow rotation if the token is locked
+		if (this.options.restrictPlayerMove) return; // don't allow rotation if the token is locked
```
Comment (cyruzzo) [inline Token.js:111]: I think this should be if (!window.DM && this.options.restrictPlayerMove ) 

so this block only applies to non DM

File: Token.js [hunk 2]
```diff
@@ -142,6 +143,7 @@ class Token {
 	}
 	move(top, left) {
 		if (this.options.locked) return; // don't allow moving if the token is locked
+		if (this.options.restrictPlayerMove) return; // don't allow moving if the token is locked
```
Comment (cyruzzo) [inline Token.js:146]: same as before... you cannot just stop movement if the option is set..   you need to check if window.DM

File: Token.js [hunk 3]
```diff
@@ -664,13 +666,22 @@ class Token {
 				old.find("img").removeClass("token-round");
 			}
 			
-			if(this.options.locked){
+			if(this.options.locked && !this.options.restrictPlayerMove){
```
Comment (cyruzzo) [inline Token.js:669]: also here
if (this.options.locked  ||   (!window.DM && this.options.restrictPlayerMove)

File: Token.js [hunk 4]
```diff
@@ -1036,6 +1068,10 @@ class Token {
 				tok.draggable("disable");
 				tok.removeClass("ui-state-disabled");
 			}
+			if(this.options.restrictPlayerMove){
```
Comment (cyruzzo) [inline Token.js:1071]: this should only apply if !window.DM ?

**Review by cyruzzo — COMMENTED** [review]
didn't check througly.. there are probably more places where the conditions need to be checked

**Comment by cyelis1224** [comment]
I'm pretty sure this is good to go now, but that's up to you. Thanks again for your help!

**Pull request merged**